src/Catty/Functions/Math/PowFunction.swift

```diff
@@ -0,0 +1,50 @@
+/**
+ *  Copyright (C) 2010-2018 The Catrobat Team
+ *  (http://developer.catrobat.org/credits)
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  An additional term exception under section 7 of the GNU Affero
+ *  General Public License, version 3, is available at
+ *  (http://developer.catrobat.org/license_additional_term)
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see http://www.gnu.org/licenses/.
+ */
+
+class PowFunction: DoubleParameterFunction {
+    static var tag = "POW"
+    static var name = "power"
+    static var defaultValue = 8.0
+    static var requiredResource = ResourceType.noResources
+    static var isIdempotent = true
+    static let position = 160
+    
+    static func firstParameter() -> FunctionParameter {
+        return .number(defaultValue: 2)
+    }
+    
+    static func secondParameter() -> FunctionParameter {
+        return .number(defaultValue: 3)
+    }
+    
+    func value(firstParameter: AnyObject?, secondParameter: AnyObject?) -> Double {
+        guard let base = firstParameter as? Double,
+            let exponent = secondParameter as? Double else {
+                return type(of: self).defaultValue
```

Comment:
`pow(2, 'abc')` would evaluate to 8, but should evaluate to 0. Please have a look at the defaultValue.